File: 3110-Score-Of-A-String/main.go
func scoreOfString(s string) int {
    
    n := len(s)
    var score int = 0

    for i := 0; i < n -1; i++{
        score += abs(int(s[i]) - int(s[i+1])) 
    }

    return score
}

func abs(x int) int {
    if x < 0 {
        return -x
    }
    return x
}

File: 3110-score-of-a-string/main.rs
impl Solution {
    pub fn score_of_string(s: String) -> i32 {
        
        let n = s.len();
        let mut score = 0;

        for i in 0..(n-1){
            let current_digit = s.chars().nth(i).unwrap() as i32 - '0' as i32;
            let next_digit = s.chars().nth(i+1).unwrap() as i32 - '0' as i32;
            score += (current_digit - next_digit).abs();
        }

        score
    }

}
